challenge: seeding

Sam09040 opened the pull request:
using snaplet for seeding and faker for mocking the data

Sam09040 committed:
initial setup

- @snaplet/seed and @snaplet/copycat dependencies
- seeds file
- seeding configuration

Sam09040 committed:
functioning seeding

## updates:
- added faker for mocking data;
- seeding keeps the first user while erasing the rest;

racsandoval commented on snaplet/seed/seed.ts:
da forma como está aqui ele sempre roda os seeds quando o arquivo é executado, potencialmente toda vez que o servidor roda, seria uma boa rodar os seeds em um comando do package json

Sam09040 replied:
é o que eu tô fazendo. os seeds tem um comando específico no package json, além de estar fora do src, então não é rodado pelo comando de start.

racsandoval replied:
isso considerando que a função `seedUsers` não é exportada em nenhum outro arquivo que é utilizado no servidor, pois caso seja ele vai executar essas linhas de código:
```
const prisma = new PrismaClient();
const existingUser = await prisma.user.findUnique({ where: { id: 1 } });

const seed = await createSeedClient({
    dryRun: false
});

await seed.$resetDatabase(["! _prisma_migrations", "! public.User"]);
```

e `seedUsers();`

Sam09040 replied:
isso, exatamente. depois de um tempo eu entendi o que você quis dizer, justamente por ter passado pelo mesmo em outra situação. vou corrigir isso.

Sam09040 committed:
separated function from seeds file

## added:
- defineSeed.ts to export the seedUsers function;

## updated:
- seed.ts to just run the seedUsers function;
